manifests/install.pp

```diff
@@ -15,13 +15,16 @@
 define kmod::install(
   $ensure=present,
   $command='/bin/true',
-  $file='/etc/modprobe.d/modprobe.conf',
+  $file="/etc/modprobe.conf/${title}.conf",
 ) {
-  kmod::generic {"install ${name}":
-    ensure  => $ensure,
-    type    => 'install',
-    module  => $name,
-    command => $command,
-    file    => $file,
+
+  kmod::setting { "kmod::install ${title}":
+    ensure   => $ensure,
+    module   => $title,
```

Comment:
Same here. In general, I'd use `$title` in resource titles and `$name` in parameters.